### DISCUSSION TITLE
A bug I encountered.

### DISCUSSION THREAD

**Discussion by TheVampireProgrammer** [sinshu/managed-doom#32] [Q&A]
You see, I am running this on a raspberry pi 4 with Debian version buster. I compiled Doom.cs and got this:
Doom.cs(20,19): error CS0234: The type or namespace name `Audio' does not exist in the namespace `ManagedDoom'. Are you missing an assembly reference?
Doom.cs(21,19): error CS0234: The type or namespace name `Video' does not exist in the namespace `ManagedDoom'. Are you missing an assembly reference?
Doom.cs(22,19): error CS0234: The type or namespace name `UserInput' does not exist in the namespace `ManagedDoom'. Are you missing an assembly reference?
Doom.cs(28,17): error CS0246: The type or namespace name `CommandLineArgs' could not be found. Are you missing an assembly reference?
Doom.cs(29,17): error CS0246: The type or namespace name `Config' could not be found. Are you missing an assembly reference?
Doom.cs(30,17): error CS0246: The type or namespace name `GameContent' could not be found. Are you missing an assembly reference?
Doom.cs(31,17): error CS0246: The type or namespace name `IVideo' could not be found. Are you missing an assembly reference?
Doom.cs(32,17): error CS0246: The type or namespace name `ISound' could not be found. Are you missing an assembly reference?
Doom.cs(33,17): error CS0246: The type or namespace name `IMusic' could not be found. Are you missing an assembly reference?
Doom.cs(34,17): error CS0246: The type or namespace name `IUserInput' could not be found. Are you missing an assembly reference?
Doom.cs(36,22): error CS0246: The type or namespace name `DoomEvent' could not be found. Are you missing an assembly reference?
Doom.cs(38,17): error CS0246: The type or namespace name `GameOptions' could not be found. Are you missing an assembly reference?
Doom.cs(40,17): error CS0246: The type or namespace name `DoomMenu' could not be found. Are you missing an assembly reference?
Doom.cs(42,17): error CS0246: The type or namespace name `OpeningSequence' could not be found. Are you missing an assembly reference?
Doom.cs(44,17): error CS0246: The type or namespace name `DemoPlayback' could not be found. Are you missing an assembly reference?
Doom.cs(46,17): error CS0246: The type or namespace name `TicCmd' could not be found. Are you missing an assembly reference?
Doom.cs(47,17): error CS0246: The type or namespace name `DoomGame' could not be found. Are you missing an assembly reference?
Doom.cs(49,17): error CS0246: The type or namespace name `WipeEffect' could not be found. Are you missing an assembly reference?
Doom.cs(52,17): error CS0246: The type or namespace name `DoomState' could not be found. Are you missing an assembly reference?
Doom.cs(53,17): error CS0246: The type or namespace name `DoomState' could not be found. Are you missing an assembly reference?
Doom.cs(63,21): error CS0246: The type or namespace name `CommandLineArgs' could not be found. Are you missing an assembly reference?
Doom.cs(63,43): error CS0246: The type or namespace name `Config' could not be found. Are you missing an assembly reference?
Doom.cs(63,58): error CS0246: The type or namespace name `GameContent' could not be found. Are you missing an assembly reference?
Doom.cs(63,79): error CS0246: The type or namespace name `IVideo' could not be found. Are you missing an assembly reference?
Doom.cs(63,93): error CS0246: The type or namespace name `ISound' could not be found. Are you missing an assembly reference?
Doom.cs(63,107): error CS0246: The type or namespace name `IMusic' could not be found. Are you missing an assembly reference?
Doom.cs(63,121): error CS0246: The type or namespace name `IUserInput' could not be found. Are you missing an assembly reference?
Doom.cs(180,29): error CS0246: The type or namespace name `GameSkill' could not be found. Are you missing an assembly reference?
Doom.cs(235,39): error CS0246: The type or namespace name `DoomKey' could not be found. Are you missing an assembly reference?
Doom.cs(356,16): error CS0246: The type or namespace name `UpdateResult' could not be found. Are you missing an assembly reference?
Doom.cs(538,31): error CS0246: The type or namespace name `DoomEvent' could not be found. Are you missing an assembly reference?
Doom.cs(546,16): error CS0246: The type or namespace name `DoomState' could not be found. Are you missing an assembly reference?
Doom.cs(547,16): error CS0246: The type or namespace name `OpeningSequence' could not be found. Are you missing an assembly reference?
Doom.cs(548,16): error CS0246: The type or namespace name `DemoPlayback' could not be found. Are you missing an assembly reference?
Doom.cs(549,16): error CS0246: The type or namespace name `GameOptions' could not be found. Are you missing an assembly reference?
Doom.cs(550,16): error CS0246: The type or namespace name `DoomGame' could not be found. Are you missing an assembly reference?
Doom.cs(551,16): error CS0246: The type or namespace name `DoomMenu' could not be found. Are you missing an assembly reference?
Doom.cs(552,16): error CS0246: The type or namespace name `WipeEffect' could not be found. Are you missing an assembly reference?

Any way to help?

**Comment by sinshu** [comment]
Sorry, I'm not familiar with environments other than Windows, so I can't give you good advice. However, I think it's unusual to try to compile Doom.cs directly. Normally, you would use the `dotnet build` command in the directory where `ManagedDoom.csproj` is located.

**Comment by IceReaper** [comment]
Checkout full project or download as zip. Navigate to folder with .sln. type "dotnet build".

**Reply by TheVampireProgrammer** [reply]
It didn't work. Know of any other solutions. I forgot to tell you I am using .Net 17.7.4

**Comment by TheVampireProgrammer** [comment]
It gave me this:
MSBuild version 17.7.4+3ebbd7c49 for .NET
  Determining projects to restore...
  Failed to download package 'Microsoft.NET.Test.Sdk.17.5.0' from 'https://api.nuget.org/v3-flatcontainer/microsoft.net.test.sdk/17.5.0/microsoft.net.test.sdk.17.5.0.nupkg'.
  An error occurred while sending the request.
    Unable to read data from the transport connection: Connection reset by peer.
    Connection reset by peer
  Failed to download package 'coverlet.collector.3.2.0' from 'https://api.nuget.org/v3-flatcontainer/coverlet.collector/3.2.0/coverlet.collector.3.2.0.nupkg'.
  An error occurred while sending the request.
    Unable to read data from the transport connection: Connection reset by peer.
    Connection reset by peer
  Failed to download package 'Ultz.Native.GLFW.3.3.9.1' from 'https://api.nuget.org/v3-flatcontainer/ultz.native.glfw/3.3.9.1/ultz.native.glfw.3.3.9.1.nupkg'.
  An error occurred while sending the request.
    Unable to read data from the transport connection: Connection reset by peer.
    Connection reset by peer
  Retrying 'FindPackagesByIdAsync' for source 'https://api.nuget.org/v3-flatcontainer/system.memory/index.json'.
  An error occurred while sending the request.
    Unable to read data from the transport connection: Connection reset by peer.
    Connection reset by peer
/home/pi/DOOM/managed-doom-master/ManagedDoom/ManagedDoom.csproj : error NU1102: Unable to find package runtime.linux-arm.Microsoft.DotNet.ILCompiler with version (= 7.0.18) [/home/pi/DOOM/managed-doom-master/ManagedDoom.sln]
/home/pi/DOOM/managed-doom-master/ManagedDoom/ManagedDoom.csproj : error NU1102:   - Found 4 version(s) in nuget.org [ Nearest version: 9.0.0-preview.1.24080.9 ] [/home/pi/DOOM/managed-doom-master/ManagedDoom.sln]
  Failed to restore /home/pi/DOOM/managed-doom-master/ManagedDoom/ManagedDoom.csproj (in 2.28 min).
^CAttempting to cancel the build...
/home/pi/.dotnet/sdk/7.0.408/NuGet.targets(158,5): error : Restore canceled! [/home/pi/DOOM/managed-doom-master/ManagedDoom.sln]

Build FAILED.

/home/pi/DOOM/managed-doom-master/ManagedDoom/ManagedDoom.csproj : error NU1102: Unable to find package runtime.linux-arm.Microsoft.DotNet.ILCompiler with version (= 7.0.18) [/home/pi/DOOM/managed-doom-master/ManagedDoom.sln]
/home/pi/DOOM/managed-doom-master/ManagedDoom/ManagedDoom.csproj : error NU1102:   - Found 4 version(s) in nuget.org [ Nearest version: 9.0.0-preview.1.24080.9 ] [/home/pi/DOOM/managed-doom-master/ManagedDoom.sln]
/home/pi/.dotnet/sdk/7.0.408/NuGet.targets(158,5): error : Restore canceled! [/home/pi/DOOM/managed-doom-master/ManagedDoom.sln]

**Comment by IceReaper** [comment]
There is no .net 17. Latest is 8. You are likely using visual studio with msbuild 17? You don't need visual studio at all. Just the .net sdk. Then you can simply compile from command line without the visual studio and msbuild Overhead

**Comment by sinshu** [comment]
According to the logs, it seems that the NuGet package download is failing due to the lack of an internet connection. How about reviewing the internet connection settings? 🤔
